robot_smach_states/examples/example_hear_grasp.py

```diff
@@ -9,33 +9,32 @@
 class Hear(smach.State):
     def __init__(self, robot):
         smach.State.__init__(self, outcomes=['succeeded', 'failed', 'preempted'],
-                             output_keys=['heared'])
+                             output_keys=['heard'])
         self.robot = robot
 
     def execute(self, userdata):
-        result = self.robot.ears.recognize(spec='Bring me a <drink>',
-                                           choices={'drink': ['coke', 'fanta']},
-                                           time_out=rospy.Duration(60))
-        choices = result.choices['drink'] if result else None
-        if choices:
-            rospy.loginfo('I heared:' + repr(choices))
-        userdata.heared = choices
+        answer = self.robot.hmi.query(description='can i have a fanta?',
+                                      grammar="T -> DET coke | DET fanta \n DET -> a | an ",
+                                      target='T',
+                                      timeout=60)
+        if answer:
+            rospy.loginfo('I heard:' + answer.sentence)
```

Comment:
As this is an example: meant to showcase the skill. I would make the robot say this out loud. Now you can only see this in a terminal.